### ISSUE TITLE
Icecream daemon bug - usr/bin/as is not executable

### ISSUE THREAD

**Issue by pold500** [icecc/icecream#184]
While compiling using icecream daemon and network, following error were encountered multiple times while trying to build a project using iceccd.

Here are the message from compile node:

Linking CXX static library xxx.a
[ 15%] Building CXX object src/components/protocol/CMakeFiles/XXXXX.dir/src/type.cc.o
ICECC[30085] 18:42:30: /var/cache/icecc/target=x86_64/7cae87a2d447f24d9e0bb8b2106a29eb/usr/bin/as is not executable - compiled on 172.18.0.1
ICECC[30085] 18:42:30: got exception 23 (172.18.0.1) 

Any fix to this?

This message occurs while trying to build any .cc file on my local node, on remote node everything goes well.

**Comment by HenryMiller1** [comment]
I have seen this when the remote doesn’t have enough disk space (in /var).  Otherwise it is a generic “Something went wrong transferring your compiler environment to the remote” error and it is another case where we need to put better logging in icecc to figure out why it happens so often.

**Comment by pold500** [comment]
@HenryMiller1 the case is, it doesn't happen on remote build machine, it happens on my node, where I run the scheduler and iceccd. Remote node building is OK.
Better logging would be great. Any chance to fix that ASAP?

**Comment by HenryMiller1** [comment]
Hmm.. Is this the same is issue #128? Patch #129 solves it for me, but isn't considered the correct fix.  If you can come up with a better fix I'd love to see it.

As for logging, it needs someone to take the time...

**Comment by mbitsnbites** [comment]
This very issue comes up often on our small network of about 15 nodes (running Ubuntu 16.04 / ICECC v1.0.1):
<pre>
ICECC[27103] 14:48:42: /var/cache/icecc/target=x86_64/d0b3c81d50b041b763d977be5952cb27/usr/bin/as is not executable - compiled on xx.yy.zz.ww
ICECC[27103] 14:48:42: got exception 23 (xx.yy.zz.ww)
</pre>

The nodes that fail will vary fairly randomly (some fail more often than others, though).

What I have seen is that:
1. The problem is usually fixed by `sudo service iceccd restart`.
2. On the failing node the corresponding folder (e.g. `/var/cache/icecc/target=x86_64/d0b3c81d50b041b763d977be5952cb27`) may exist, but it's empty, even if there is plenty of space left on the disk.

This leads me to believe that there is some intermittent failure in the toolchain cache (either during delivery, or if it's purged prematurely or similar).

**Comment by HenryMiller1** [comment]
@mbitsnbites please try the latest master. The ubuntu packages are very old.  There is better logging in master.  I'm running the latest master and I do not see this anymore - though I cannot guarantee that this isn't coincidence.

Note that the ubuntu pakages do not properly save log messages which makes this hard to debug. See  https://bugs.launchpad.net/ubuntu/+source/icecc/+bug/1671926

**Comment by mbitsnbites** [comment]
@HenryMiller1 What is the version interoperability situation for ICECC? Could I try a later build together with a scheduler and other clients that run 1.0.1?

**Comment by HenryMiller1** [comment]
@mbitsnbites we do a reasonable job about handling backward compatibility.  Generally I would expect that everything will work.

The only exception is if you use debug-fission. Debug-fission is not supported at all in 1.0.1, so this probably isn't an issue for you, but if it is you need to upgrade the scheduler and several other clients.  (debug fission requires newer clients)

**Comment by ericch1** [comment]
Hi,
First, thanks a lot for the gook work at maintaining icecream again!  We are using it on a day by day basis since a lot of years!

Now, I am using the latest master snapshot (6a5fa3) on my workstation+sheduler+remote and I have this error:

ICECC[98168] 11:26:01: /var/cache/icecream/target=x86_64/e63cdd490130b54ce9a37e4b45fed464/usr/bin/as is not executable - compiled on 132.203.xxx.yyy
ICECC[98168] 11:26:01: got exception 23 (132.203.xxx.yyy) 

(i changed the end of the ip for xxx.yyy)

How can I help to debug this?

Thanks,
Eric

**Comment by ericch1** [comment]
> Is this issue #128?
I think not, since the problem is on a remote, and #128 stats that build on local computer fails

> Is this a variation of issue #245?
hmmm, I don't know, but how could this happen on *some* remote.  Also, the failling remote change over time...

> Is /var mounted as noexec or some such option on the remote machine?
It is the "/" partition:

cd /var/cache/icecream/
[root@roederer icecream]# df .
Filesystem     1K-blocks     Used Available Use% Mounted on
/dev/sda1       50264772 21248152  26440236  45% /

ls -la .
total 16
drwxrwxr-x.  4 icecream icecream 4096 May 25 13:59 .
drwxr-xr-x. 18 root     root     4096 May  6 08:57 ..
drwxrwxr-x   2 icecream icecream 4096 May 25 13:59 native
drwxrwx---   2 icecream icecream 4096 May 25 13:59 'target=x86_64'

Is it normal that the directory name is in between single quotes?

> Is the remote machine x86_64?

Yes, it is x86_64:
-bash-4.3$ uname -a
Linux roederer 4.10.15-200.fc25.x86_64 #1 SMP Mon May 8 18:46:06 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

> If the above are not an issue ....

Ok, If I create the environment, it is strangley very long and have some warning at end:

Here is the complete (and very long) output:

cd /tmp
[eric@lorien] tmp> /opt/icecream-1.1/icecc-create-env  --clang /usr/bin/clang 
bash: /opt/icecream-1.1/icecc-create-env: No such file or directory
[eric@lorien] tmp> /opt/icecream-1.1/bin/icecc-create-env  --clang /usr/bin/clang
adding file /bin/true
adding file /lib64/libc.so.6
adding file /lib64/ld-linux-x86-64.so.2
adding file /usr/bin/clang
adding file /usr/bin/../lib64/libLLVMAArch64CodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMAArch64AsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMAArch64Utils.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMSupport.so.3.9
adding file /lib64/libdl.so.2
adding file /lib64/libtinfo.so.5
adding file /lib64/libpthread.so.0
adding file /lib64/libz.so.1
adding file /lib64/libm.so.6
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libstdc++.so.6
adding file /lib64/libgcc_s.so.1
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libstdc++.so.6
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMMC.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMSupport.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libstdc++.so.6
adding file /usr/bin/../lib64/../lib64/libLLVMAArch64Desc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMAArch64AsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMAArch64Utils.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMSupport.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libstdc++.so.6
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMMC.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMAArch64Info.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMAArch64Info.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMAArch64Utils.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMAnalysis.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMCore.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMProfileData.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMCore.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMAnalysis.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMProfileData.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMCore.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMSupport.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libstdc++.so.6
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMAnalysis.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMProfileData.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMCore.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMSupport.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libstdc++.so.6
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMBitReader.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMBitWriter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMAnalysis.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMProfileData.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMCore.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMSupport.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libstdc++.so.6
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMScalarOpts.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMInstCombine.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMAnalysis.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMProfileData.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMCore.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMSupport.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libstdc++.so.6
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMTransformUtils.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMAnalysis.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMProfileData.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMCore.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMSupport.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libstdc++.so.6
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMTransformUtils.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMTarget.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMMC.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMTransformUtils.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMDebugInfoCodeView.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMMCParser.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMTarget.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMBitReader.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMBitWriter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMScalarOpts.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMInstCombine.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMTransformUtils.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMCore.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMGlobalISel.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMMC.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMScalarOpts.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMInstCombine.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMSelectionDAG.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMSupport.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMTarget.so.3.9
adding file /usr/bin/../lib64/../lib64/libstdc++.so.6
adding file /usr/bin/../lib64/libLLVMAArch64AsmParser.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMMCParser.so.3.9
adding file /usr/bin/../lib64/libLLVMAArch64Desc.so.3.9
adding file /usr/bin/../lib64/libLLVMAArch64Info.so.3.9
adding file /usr/bin/../lib64/libLLVMAMDGPUCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMAMDGPUAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMAMDGPUUtils.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMAMDGPUDesc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMAMDGPUAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMAMDGPUUtils.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMAMDGPUInfo.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMAMDGPUInfo.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMAMDGPUUtils.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMTransformUtils.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMVectorize.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMipo.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMIRReader.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMAsmParser.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMInstrumentation.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMLinker.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMObject.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMMCParser.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMVectorize.so.3.9
adding file /usr/bin/../lib64/libLLVMAMDGPUAsmParser.so.3.9
adding file /usr/bin/../lib64/libLLVMAMDGPUDesc.so.3.9
adding file /usr/bin/../lib64/libLLVMAMDGPUInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMARMCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMARMAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMARMDesc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMARMAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMARMInfo.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMMCDisassembler.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMARMInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMARMAsmParser.so.3.9
adding file /usr/bin/../lib64/libLLVMARMDesc.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMMCDisassembler.so.3.9
adding file /usr/bin/../lib64/libLLVMARMInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMBPFCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMBPFDesc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMBPFAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMBPFInfo.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMBPFInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMBPFDesc.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMBPFAsmPrinter.so.3.9
adding file /usr/bin/../lib64/libLLVMBPFInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMHexagonCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMHexagonDesc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMHexagonInfo.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMHexagonInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMHexagonAsmParser.so.3.9
adding file /usr/bin/../lib64/libLLVMHexagonDesc.so.3.9
adding file /usr/bin/../lib64/libLLVMHexagonInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMMipsCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMMipsAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMMipsDesc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMMipsAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMMipsInfo.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMMipsInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMMipsAsmParser.so.3.9
adding file /usr/bin/../lib64/libLLVMMipsDesc.so.3.9
adding file /usr/bin/../lib64/libLLVMMipsInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMMSP430CodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMMSP430AsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMMSP430Desc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMMSP430AsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMMSP430Info.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMMSP430Info.so.3.9
adding file /usr/bin/../lib64/libLLVMMSP430Desc.so.3.9
adding file /usr/bin/../lib64/libLLVMMSP430Info.so.3.9
adding file /usr/bin/../lib64/libLLVMNVPTXCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMNVPTXAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMNVPTXDesc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMNVPTXAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMNVPTXInfo.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMNVPTXInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMNVPTXDesc.so.3.9
adding file /usr/bin/../lib64/libLLVMNVPTXInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMPowerPCCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMPowerPCAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMPowerPCDesc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMPowerPCAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMPowerPCInfo.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMPowerPCInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMPowerPCAsmParser.so.3.9
adding file /usr/bin/../lib64/libLLVMPowerPCDesc.so.3.9
adding file /usr/bin/../lib64/libLLVMPowerPCInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMSparcCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMSparcAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMSparcDesc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMSparcAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMSparcInfo.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMSparcInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMSparcAsmParser.so.3.9
adding file /usr/bin/../lib64/libLLVMSparcDesc.so.3.9
adding file /usr/bin/../lib64/libLLVMSparcInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMSystemZCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMSystemZAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMSystemZDesc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMSystemZAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMSystemZInfo.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMSystemZInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMSystemZAsmParser.so.3.9
adding file /usr/bin/../lib64/libLLVMSystemZDesc.so.3.9
adding file /usr/bin/../lib64/libLLVMSystemZInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMX86CodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMX86AsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMX86Utils.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMX86Desc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMX86AsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMX86Utils.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMX86Info.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMX86Info.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMX86Utils.so.3.9
adding file /usr/bin/../lib64/libLLVMX86AsmParser.so.3.9
adding file /usr/bin/../lib64/libLLVMX86Desc.so.3.9
adding file /usr/bin/../lib64/libLLVMX86Info.so.3.9
adding file /usr/bin/../lib64/libLLVMXCoreCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMXCoreAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMXCoreDesc.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMXCoreAsmPrinter.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMXCoreInfo.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMXCoreInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMXCoreDesc.so.3.9
adding file /usr/bin/../lib64/libLLVMXCoreInfo.so.3.9
adding file /usr/bin/../lib64/libLLVMAnalysis.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMProfileData.so.3.9
adding file /usr/bin/../lib64/libLLVMCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMBitReader.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMBitWriter.so.3.9
adding file /usr/bin/../lib64/libLLVMCore.so.3.9
adding file /usr/bin/../lib64/libLLVMipo.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMIRReader.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMAsmParser.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMInstCombine.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMInstrumentation.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMLinker.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMObject.so.3.9
adding file /usr/bin/../lib64/libLLVMInstCombine.so.3.9
adding file /usr/bin/../lib64/libLLVMInstrumentation.so.3.9
adding file /usr/bin/../lib64/libLLVMMC.so.3.9
adding file /usr/bin/../lib64/libLLVMMCParser.so.3.9
adding file /usr/bin/../lib64/libLLVMObjCARCOpts.so.3.9
adding file /usr/bin/../lib64/libLLVMOption.so.3.9
adding file /usr/bin/../lib64/libLLVMScalarOpts.so.3.9
adding file /usr/bin/../lib64/libLLVMSupport.so.3.9
adding file /usr/bin/../lib64/libLLVMTransformUtils.so.3.9
adding file /usr/bin/../lib64/libLLVMVectorize.so.3.9
adding file /usr/bin/../lib64/libclangBasic.so.3.9
adding file /usr/bin/../lib64/libclangCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangAST.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangBasic.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangLex.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libclangBasic.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangBasic.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangFrontend.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangAST.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libclangLex.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libclangBasic.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMMC.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangDriver.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMOption.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangEdit.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libclangAST.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libclangLex.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libclangBasic.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMMC.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangParse.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libclangSema.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libclangAST.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libclangLex.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libclangBasic.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMMC.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libclangAnalysis.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libclangAST.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libclangLex.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libclangBasic.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMMC.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libclangEdit.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangSema.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libclangAnalysis.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libclangEdit.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangSerialization.so.3.9

adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMOption.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangLex.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMCoverage.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMObjCARCOpts.so.3.9
adding file /usr/bin/../lib64/libclangDriver.so.3.9
adding file /usr/bin/../lib64/../lib64/libLLVMOption.so.3.9
adding file /usr/bin/../lib64/libclangFrontend.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangDriver.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangEdit.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangParse.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangSema.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangAnalysis.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangSerialization.so.3.9
adding file /usr/bin/../lib64/libclangFrontendTool.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangCodeGen.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangFrontend.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libclangDriver.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMOption.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libclangParse.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libclangSema.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libclangAnalysis.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libclangAST.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libclangLex.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libclangBasic.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMCore.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMSupport.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libstdc++.so.6
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMMC.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/libclangEdit.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMMCParser.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libclangSerialization.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMBitReader.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMCoverage.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMObject.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMipo.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMIRReader.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libLLVMAsmParser.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMInstrumentation.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMLinker.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libLLVMVectorize.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libLLVMObjCARCOpts.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangRewriteFrontend.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangRewrite.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangARCMigrate.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangStaticAnalyzerCheckers.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libclangStaticAnalyzerCore.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/libclangRewrite.so.3.9
adding file /usr/bin/../lib64/../lib64/libclangStaticAnalyzerFrontend.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/libclangStaticAnalyzerCore.so.3.9
adding file /usr/bin/../lib64/../lib64/../lib64/../lib64/libclangRewrite.so.3.9
adding file /usr/bin/../lib64/libstdc++.so.6
adding file /usr/bin/gcc=/opt/icecream-1.1/lib/icecc/compilerwrapper
adding file /usr/lib64/libstdc++.so.6
adding file /usr/bin/g++=/opt/icecream-1.1/lib/icecc/compilerwrapper
adding file /usr/bin/as
adding file /usr/lib64/libopcodes-2.26.1.so
adding file /usr/lib64/libbfd-2.26.1.so
adding file /usr/lib64/clang/3.9.1/include/mwaitxintrin.h=/usr/bin/../lib64/clang/3.9.1/include/mwaitxintrin.h
adding file /usr/lib64/clang/3.9.1/include/__clang_cuda_cmath.h=/usr/bin/../lib64/clang/3.9.1/include/__clang_cuda_cmath.h
adding file /usr/lib64/clang/3.9.1/include/nmmintrin.h=/usr/bin/../lib64/clang/3.9.1/include/nmmintrin.h
adding file /usr/lib64/clang/3.9.1/include/__clang_cuda_intrinsics.h=/usr/bin/../lib64/clang/3.9.1/include/__clang_cuda_intrinsics.h
adding file /usr/lib64/clang/3.9.1/include/__clang_cuda_math_forward_declares.h=/usr/bin/../lib64/clang/3.9.1/include/__clang_cuda_math_forward_declares.h
adding file /usr/lib64/clang/3.9.1/include/msa.h=/usr/bin/../lib64/clang/3.9.1/include/msa.h
adding file /usr/lib64/clang/3.9.1/include/__clang_cuda_runtime_wrapper.h=/usr/bin/../lib64/clang/3.9.1/include/__clang_cuda_runtime_wrapper.h
adding file /usr/lib64/clang/3.9.1/include/omp.h=/usr/bin/../lib64/clang/3.9.1/include/omp.h
adding file /usr/lib64/clang/3.9.1/include/__stddef_max_align_t.h=/usr/bin/../lib64/clang/3.9.1/include/__stddef_max_align_t.h
adding file /usr/lib64/clang/3.9.1/include/stdarg.h=/usr/bin/../lib64/clang/3.9.1/include/stdarg.h
adding file /usr/lib64/clang/3.9.1/include/__wmmintrin_aes.h=/usr/bin/../lib64/clang/3.9.1/include/__wmmintrin_aes.h
adding file /usr/lib64/clang/3.9.1/include/opencl-c.h=/usr/bin/../lib64/clang/3.9.1/include/opencl-c.h
adding file /usr/lib64/clang/3.9.1/include/__wmmintrin_pclmul.h=/usr/bin/../lib64/clang/3.9.1/include/__wmmintrin_pclmul.h
adding file /usr/lib64/clang/3.9.1/include/adxintrin.h=/usr/bin/../lib64/clang/3.9.1/include/adxintrin.h
adding file /usr/lib64/clang/3.9.1/include/altivec.h=/usr/bin/../lib64/clang/3.9.1/include/altivec.h
adding file /usr/lib64/clang/3.9.1/include/ammintrin.h=/usr/bin/../lib64/clang/3.9.1/include/ammintrin.h
adding file /usr/lib64/clang/3.9.1/include/arm_acle.h=/usr/bin/../lib64/clang/3.9.1/include/arm_acle.h
adding file /usr/lib64/clang/3.9.1/include/arm_neon.h=/usr/bin/../lib64/clang/3.9.1/include/arm_neon.h
adding file /usr/lib64/clang/3.9.1/include/stdatomic.h=/usr/bin/../lib64/clang/3.9.1/include/stdatomic.h
adding file /usr/lib64/clang/3.9.1/include/avx2intrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx2intrin.h
adding file /usr/lib64/clang/3.9.1/include/stdbool.h=/usr/bin/../lib64/clang/3.9.1/include/stdbool.h
adding file /usr/lib64/clang/3.9.1/include/avx512bwintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512bwintrin.h
adding file /usr/lib64/clang/3.9.1/include/stddef.h=/usr/bin/../lib64/clang/3.9.1/include/stddef.h
adding file /usr/lib64/clang/3.9.1/include/avx512cdintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512cdintrin.h
adding file /usr/lib64/clang/3.9.1/include/stdint.h=/usr/bin/../lib64/clang/3.9.1/include/stdint.h
adding file /usr/lib64/clang/3.9.1/include/avx512dqintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512dqintrin.h
adding file /usr/lib64/clang/3.9.1/include/stdnoreturn.h=/usr/bin/../lib64/clang/3.9.1/include/stdnoreturn.h
adding file /usr/lib64/clang/3.9.1/include/avx512erintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512erintrin.h
adding file /usr/lib64/clang/3.9.1/include/tbmintrin.h=/usr/bin/../lib64/clang/3.9.1/include/tbmintrin.h
adding file /usr/lib64/clang/3.9.1/include/avx512fintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512fintrin.h
adding file /usr/lib64/clang/3.9.1/include/tgmath.h=/usr/bin/../lib64/clang/3.9.1/include/tgmath.h
adding file /usr/lib64/clang/3.9.1/include/avx512ifmaintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512ifmaintrin.h
adding file /usr/lib64/clang/3.9.1/include/shaintrin.h=/usr/bin/../lib64/clang/3.9.1/include/shaintrin.h
adding file /usr/lib64/clang/3.9.1/include/avx512ifmavlintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512ifmavlintrin.h
adding file /usr/lib64/clang/3.9.1/include/tmmintrin.h=/usr/bin/../lib64/clang/3.9.1/include/tmmintrin.h
adding file /usr/lib64/clang/3.9.1/include/avx512pfintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512pfintrin.h
adding file /usr/lib64/clang/3.9.1/include/unwind.h=/usr/bin/../lib64/clang/3.9.1/include/unwind.h
adding file /usr/lib64/clang/3.9.1/include/avx512vbmiintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512vbmiintrin.h
adding file /usr/lib64/clang/3.9.1/include/smmintrin.h=/usr/bin/../lib64/clang/3.9.1/include/smmintrin.h
adding file /usr/lib64/clang/3.9.1/include/avx512vbmivlintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512vbmivlintrin.h
adding file /usr/lib64/clang/3.9.1/include/vadefs.h=/usr/bin/../lib64/clang/3.9.1/include/vadefs.h
adding file /usr/lib64/clang/3.9.1/include/avx512vlbwintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512vlbwintrin.h
adding file /usr/lib64/clang/3.9.1/include/varargs.h=/usr/bin/../lib64/clang/3.9.1/include/varargs.h
adding file /usr/lib64/clang/3.9.1/include/avx512vlcdintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512vlcdintrin.h
adding file /usr/lib64/clang/3.9.1/include/vecintrin.h=/usr/bin/../lib64/clang/3.9.1/include/vecintrin.h
adding file /usr/lib64/clang/3.9.1/include/avx512vldqintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512vldqintrin.h
adding file /usr/lib64/clang/3.9.1/include/wmmintrin.h=/usr/bin/../lib64/clang/3.9.1/include/wmmintrin.h
adding file /usr/lib64/clang/3.9.1/include/avx512vlintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avx512vlintrin.h
adding file /usr/lib64/clang/3.9.1/include/avxintrin.h=/usr/bin/../lib64/clang/3.9.1/include/avxintrin.h
adding file /usr/lib64/clang/3.9.1/include/x86intrin.h=/usr/bin/../lib64/clang/3.9.1/include/x86intrin.h
adding file /usr/lib64/clang/3.9.1/include/bmi2intrin.h=/usr/bin/../lib64/clang/3.9.1/include/bmi2intrin.h
adding file /usr/lib64/clang/3.9.1/include/bmiintrin.h=/usr/bin/../lib64/clang/3.9.1/include/bmiintrin.h
adding file /usr/lib64/clang/3.9.1/include/xmmintrin.h=/usr/bin/../lib64/clang/3.9.1/include/xmmintrin.h
adding file /usr/lib64/clang/3.9.1/include/clflushoptintrin.h=/usr/bin/../lib64/clang/3.9.1/include/clflushoptintrin.h
adding file /usr/lib64/clang/3.9.1/include/cpuid.h=/usr/bin/../lib64/clang/3.9.1/include/cpuid.h
adding file /usr/lib64/clang/3.9.1/include/xopintrin.h=/usr/bin/../lib64/clang/3.9.1/include/xopintrin.h
adding file /usr/lib64/clang/3.9.1/include/cuda_builtin_vars.h=/usr/bin/../lib64/clang/3.9.1/include/cuda_builtin_vars.h
adding file /usr/lib64/clang/3.9.1/include/emmintrin.h=/usr/bin/../lib64/clang/3.9.1/include/emmintrin.h
adding file /usr/lib64/clang/3.9.1/include/xsavecintrin.h=/usr/bin/../lib64/clang/3.9.1/include/xsavecintrin.h
adding file /usr/lib64/clang/3.9.1/include/f16cintrin.h=/usr/bin/../lib64/clang/3.9.1/include/f16cintrin.h
adding file /usr/lib64/clang/3.9.1/include/float.h=/usr/bin/../lib64/clang/3.9.1/include/float.h
adding file /usr/lib64/clang/3.9.1/include/xsaveintrin.h=/usr/bin/../lib64/clang/3.9.1/include/xsaveintrin.h
adding file /usr/lib64/clang/3.9.1/include/fma4intrin.h=/usr/bin/../lib64/clang/3.9.1/include/fma4intrin.h
adding file /usr/lib64/clang/3.9.1/include/fmaintrin.h=/usr/bin/../lib64/clang/3.9.1/include/fmaintrin.h
adding file /usr/lib64/clang/3.9.1/include/xsaveoptintrin.h=/usr/bin/../lib64/clang/3.9.1/include/xsaveoptintrin.h
adding file /usr/lib64/clang/3.9.1/include/fxsrintrin.h=/usr/bin/../lib64/clang/3.9.1/include/fxsrintrin.h
adding file /usr/lib64/clang/3.9.1/include/htmintrin.h=/usr/bin/../lib64/clang/3.9.1/include/htmintrin.h
adding file /usr/lib64/clang/3.9.1/include/xsavesintrin.h=/usr/bin/../lib64/clang/3.9.1/include/xsavesintrin.h
adding file /usr/lib64/clang/3.9.1/include/htmxlintrin.h=/usr/bin/../lib64/clang/3.9.1/include/htmxlintrin.h
adding file /usr/lib64/clang/3.9.1/include/xtestintrin.h=/usr/bin/../lib64/clang/3.9.1/include/xtestintrin.h
adding file /usr/lib64/clang/3.9.1/include/ia32intrin.h=/usr/bin/../lib64/clang/3.9.1/include/ia32intrin.h
adding file /usr/lib64/clang/3.9.1/include/immintrin.h=/usr/bin/../lib64/clang/3.9.1/include/immintrin.h
adding file /usr/lib64/clang/3.9.1/include/intrin.h=/usr/bin/../lib64/clang/3.9.1/include/intrin.h
adding file /usr/lib64/clang/3.9.1/include/inttypes.h=/usr/bin/../lib64/clang/3.9.1/include/inttypes.h
adding file /usr/lib64/clang/3.9.1/include/iso646.h=/usr/bin/../lib64/clang/3.9.1/include/iso646.h
adding file /usr/lib64/clang/3.9.1/include/limits.h=/usr/bin/../lib64/clang/3.9.1/include/limits.h
adding file /usr/lib64/clang/3.9.1/include/lzcntintrin.h=/usr/bin/../lib64/clang/3.9.1/include/lzcntintrin.h
adding file /usr/lib64/clang/3.9.1/include/mm3dnow.h=/usr/bin/../lib64/clang/3.9.1/include/mm3dnow.h
adding file /usr/lib64/clang/3.9.1/include/mm_malloc.h=/usr/bin/../lib64/clang/3.9.1/include/mm_malloc.h
adding file /usr/lib64/clang/3.9.1/include/mmintrin.h=/usr/bin/../lib64/clang/3.9.1/include/mmintrin.h
adding file /usr/lib64/clang/3.9.1/include/module.modulemap=/usr/bin/../lib64/clang/3.9.1/include/module.modulemap
adding file /usr/lib64/clang/3.9.1/include/pkuintrin.h=/usr/bin/../lib64/clang/3.9.1/include/pkuintrin.h
adding file /usr/lib64/clang/3.9.1/include/pmmintrin.h=/usr/bin/../lib64/clang/3.9.1/include/pmmintrin.h
adding file /usr/lib64/clang/3.9.1/include/popcntintrin.h=/usr/bin/../lib64/clang/3.9.1/include/popcntintrin.h
adding file /usr/lib64/clang/3.9.1/include/prfchwintrin.h=/usr/bin/../lib64/clang/3.9.1/include/prfchwintrin.h
adding file /usr/lib64/clang/3.9.1/include/rdseedintrin.h=/usr/bin/../lib64/clang/3.9.1/include/rdseedintrin.h
adding file /usr/lib64/clang/3.9.1/include/rtmintrin.h=/usr/bin/../lib64/clang/3.9.1/include/rtmintrin.h
adding file /usr/lib64/clang/3.9.1/include/s390intrin.h=/usr/bin/../lib64/clang/3.9.1/include/s390intrin.h
adding file /usr/lib64/clang/3.9.1/include/sanitizer/allocator_interface.h=/usr/bin/../lib64/clang/3.9.1/include/sanitizer/allocator_interface.h
adding file /usr/lib64/clang/3.9.1/include/sanitizer/asan_interface.h=/usr/bin/../lib64/clang/3.9.1/include/sanitizer/asan_interface.h
adding file /usr/lib64/clang/3.9.1/include/sanitizer/common_interface_defs.h=/usr/bin/../lib64/clang/3.9.1/include/sanitizer/common_interface_defs.h
adding file /usr/lib64/clang/3.9.1/include/sanitizer/coverage_interface.h=/usr/bin/../lib64/clang/3.9.1/include/sanitizer/coverage_interface.h
adding file /usr/lib64/clang/3.9.1/include/sanitizer/dfsan_interface.h=/usr/bin/../lib64/clang/3.9.1/include/sanitizer/dfsan_interface.h
adding file /usr/lib64/clang/3.9.1/include/sanitizer/esan_interface.h=/usr/bin/../lib64/clang/3.9.1/include/sanitizer/esan_interface.h
adding file /usr/lib64/clang/3.9.1/include/sanitizer/linux_syscall_hooks.h=/usr/bin/../lib64/clang/3.9.1/include/sanitizer/linux_syscall_hooks.h
adding file /usr/lib64/clang/3.9.1/include/sanitizer/lsan_interface.h=/usr/bin/../lib64/clang/3.9.1/include/sanitizer/lsan_interface.h
adding file /usr/lib64/clang/3.9.1/include/sanitizer/msan_interface.h=/usr/bin/../lib64/clang/3.9.1/include/sanitizer/msan_interface.h
adding file /usr/lib64/clang/3.9.1/include/sanitizer/tsan_interface_atomic.h=/usr/bin/../lib64/clang/3.9.1/include/sanitizer/tsan_interface_atomic.h
adding file /usr/lib64/clang/3.9.1/include/stdalign.h=/usr/bin/../lib64/clang/3.9.1/include/stdalign.h
adding file /usr/bin/objcopy
adding file /etc/ld.so.conf=/tmp/icecc_ld_so_confKgaWlq
creating f927ab1d73809968862802c103714e1e.tar.gz
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../' from member names
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../' from member names
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../' from hard link targets
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../' from member names
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../' from hard link targets
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../' from member names
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../' from hard link targets
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../' from member names
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../' from hard link targets
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../' from member names
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../' from hard link targets
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../' from member names
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../lib64/../' from hard link targets
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../' from member names
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../lib64/../' from hard link targets
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../' from member names
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../lib64/../' from hard link targets
tar: Removing leading `usr/bin/../lib64/../lib64/../' from member names
tar: Removing leading `usr/bin/../lib64/../lib64/../lib64/../' from hard link targets
tar: Removing leading `usr/bin/../lib64/../' from member names
tar: Removing leading `usr/bin/../lib64/../lib64/../' from hard link targets
tar: Removing leading `usr/bin/../' from member names
tar: Removing leading `usr/bin/../lib64/../' from hard link targets

is this normal?

But th "as" in it is executable:
tar -xvf ../f927ab1d73809968862802c103714e1e.tar.gz 
./usr/bin/as --version
GNU assembler (GNU Binutils; openSUSE Leap 42.1) 2.26.1

> run ldd on usr/bin/as 
all is fine
> chroot iceeric /usr/bin/as --version
is fine too

> then look at the log file...

ok, here are some errors:

May 25 16:06:18 roederer iceccd: [23156] 16:06:18: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 171822

looks like the environment is not transfered or what?

Hope this help!

Thanks a lot!

Eric

**Comment by HenryMiller1** [comment]
Nothing stands out there.  the environment is long, but it looks like it is long because clang has a lot of libraries it links to and we find/install them correctly.

The message "I don't have environment" should be followed transferring the environment so I don't think that is it.

This error has been seen by multiple different people so I think it is real, but I can't figure out where to go next to figure out why things do not work.

**Comment by ericch1** [comment]
> the environment is long, but it looks like it is long because clang has a lot of libraries it links to and we find/install them correctly.

hmmm, I am worried by 3 things: 

- The same libs are repeated a lot of times, for example libclangBasic.so.3.9 appears 9 times..

- There a "warnings" like message: "tar: Removing leading `usr/bin/../lib64/../' from hard link targets"

- It takes long to generate the environment: 62 seconds!  Could this lead to any kind of timeout for the remote for receiving the environment?

Thanks,
Eric

**Comment by HenryMiller1** [comment]
We should handle the libs copied multiple times.  We check what libraries each program needs and add it - we expect that everything will link against libc for example

The removing leading usr/bin/../lib64/../ is referring to that path ultimately ends up meaning usr so we can remove bin/../lib64/..  (of course the full path is longer and isn't usr)

I don't think that is a timeout, the local generation of the environment should happen first.  However if it use, just put the environment you just generated into USE_VERSION which will ensure that the environment generation doesn't happen.

**Comment by ericch1** [comment]
I still have this issu of (ICECC[96684] 10:32:23: /var/cache/icecream/target=x86_64/a56529631d0633503a3a54be0af992f7/usr/bin/as is not executable - compiled on ...)

which is anoying...  It seems to happen when someone is computing on the "faulty" remote...  otherwise, the remote works well...

One thing about this is: why does it slow down soooooo much all the compilation ?  It seems to retry to send or wait a long time for this "faulty" remote to "return"...

Can it be accelerated?

Thanks,
Eric

**Comment by mbitsnbites** [comment]
I agree with @ericch1 - it should be possible to avoid the slow-down.

A side note: In my experience the faulty node will start working again if you restart `iceccd` on that node (e.g. `sudo service iceccd restart`).

**Comment by ericch1** [comment]
I don't kow if it will help, but in the /var/log/messages of the faulty serverthe error is repeated a lot of times in the logs:

Sep  7 13:39:51 elrond iceccd: [3926] 13:39:51: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1103555
Sep  7 13:39:51 elrond iceccd: [3927] 13:39:51: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1103558
Sep  7 13:39:51 elrond iceccd: [3928] 13:39:51: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1103559
....
Sep  7 13:39:55 elrond iceccd: [3971] 13:39:55: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1103645
Sep  7 13:39:55 elrond iceccd: [3972] 13:39:55: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1103646
Sep  7 13:39:56 elrond iceccd: [3973] 13:39:56: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1103647

then there is this message when it stops:
Sep  7 13:40:26 elrond iceccd: [3992] 13:40:26: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1103786
Sep  7 13:41:07 elrond iceccd: [4082] 13:41:07: handle_transfer_env
Sep  7 13:41:07 elrond iceccd: [4082] 13:41:07: got pid 3996
Sep  7 13:41:08 elrond iceccd: [4082] 13:41:08: handle_transfer_env_done
Sep  7 13:41:08 elrond iceccd: [4082] 13:41:08: installed_size: 77549582
Sep  7 13:41:08 elrond iceccd: [4082] 13:41:08: installed x86_64/ICECC_VERSION_clang3.8 size: 77549582 all: 105424377
Sep  7 13:41:08 elrond iceccd: [4082] 13:41:08: reannounce_environments

but on my "client" console, i got the errors:
[libdev] OptimElas créée: Thu Sep  7 13:45:03 EDT 2017
ICECC[184125] 13:39:52: /var/cache/icecream/target=x86_64/a56529631d0633503a3a54be0af992f7/usr/bin/as is not executable - compiled on 132.203.7.21

which are in fact sent *a lot* latter....: look at the delay: received 👍 
[libdev] OptimElas créée: Thu Sep  7 13:45:03 ...

then 
ICECC[184125] 13:39:52:  ...

Strange no?

Eric

**Comment by HenryMiller1** [comment]
What all these errors are saying is that your environment file is either
not transferring correctly, or it is not usable.
Do you have enough disk space? Most people do these days, but it is
possible your partition scheme doesn't leave ice cream enough space to
work with.
Failing that, Time to start looking at your environment file. Run 'tar -
tzf' on it to see the contents.
I still have not seen the client logs with ICECC_DEBUG set which makes
figuring anything out hard.
--
  Henry Miller
  hank@millerfarm.com



On Thu, Sep 7, 2017, at 12:50 PM, Eric Chamberland wrote:
> I don't kow if it will help, but in the /var/log/messages of the
> faulty serverthe error is repeated a lot of times in the logs:> Sep  7 13:39:51 elrond iceccd: [3926] 13:39:51: I don't have
> environment a56529631d0633503a3a54be0af992f7(x86_64) 1103555>  Sep  7 13:39:51 elrond iceccd: [3927] 13:39:51: I don't have
>  environment a56529631d0633503a3a54be0af992f7(x86_64) 1103558>  Sep  7 13:39:51 elrond iceccd: [3928] 13:39:51: I don't have
>  environment a56529631d0633503a3a54be0af992f7(x86_64) 1103559>  ....
>  Sep  7 13:39:55 elrond iceccd: [3971] 13:39:55: I don't have
>  environment a56529631d0633503a3a54be0af992f7(x86_64) 1103645>  Sep  7 13:39:55 elrond iceccd: [3972] 13:39:55: I don't have
>  environment a56529631d0633503a3a54be0af992f7(x86_64) 1103646>  Sep  7 13:39:56 elrond iceccd: [3973] 13:39:56: I don't have
>  environment a56529631d0633503a3a54be0af992f7(x86_64) 1103647> then there is this message when it stops:
>  Sep  7 13:40:26 elrond iceccd: [3992] 13:40:26: I don't have
>  environment a56529631d0633503a3a54be0af992f7(x86_64) 1103786>  Sep  7 13:41:07 elrond iceccd: [4082] 13:41:07: handle_transfer_env
>  Sep  7 13:41:07 elrond iceccd: [4082] 13:41:07: got pid 3996
>  Sep  7 13:41:08 elrond iceccd: [4082] 13:41:08:
>  handle_transfer_env_done>  Sep  7 13:41:08 elrond iceccd: [4082] 13:41:08: installed_size:
>  77549582>  Sep  7 13:41:08 elrond iceccd: [4082] 13:41:08: installed
>  x86_64/ICECC_VERSION_clang3.8 size: 77549582 all: 105424377>  Sep  7 13:41:08 elrond iceccd: [4082] 13:41:08:
>  reannounce_environments> but on my "client" console, i got the errors:
>  [libdev] OptimElas créée: Thu Sep  7 13:45:03 EDT 2017
>  ICECC[184125] 13:39:52: /var/cache/icecream/target=x86_64/a56529631d-
>  0633503a3a54be0af992f7/usr/bin/as is not executable - compiled on
>  132.203.7.21> which are in fact sent *a lot* latter....: look at the delay:
> received 👍>  [libdev] OptimElas créée: Thu Sep  7 13:45:03 ...


> then
>  ICECC[184125] 13:39:52:  ...


> Strange no?


> Eric


> — You are receiving this because you were mentioned. Reply to this
> email directly, view it on GitHub[1], or mute the thread[2].> 

Links:

  1. https://github.com/icecc/icecream/issues/184#issuecomment-327873403

**Comment by ericch1** [comment]
Ok I have set ICECC_DEBUG=debug.

Looking at the logs, I can see it is working right now:
<pre>
...
[18022] 12:11:04: handle_transfer_env
[18022] 12:11:04: got pid 18343
[18022] 12:11:05: handle_transfer_env_done
[18022] 12:11:05: installed_size: 87289375
[18022] 12:11:05: installed x86_64/a56529631d0633503a3a54be0af992f7 size: 87289375 all: 173484305
[18022] 12:11:05: reannounce_environments 
</pre>

But I am asking myseld: since the very same "environment" is transfered, when is it deleted or retransfered?  Can it be deleted by a cleanup process (garbage collection) just while I am sending a new request?

Thanks,
Eric

**Comment by HenryMiller1** [comment]
There is an algorithm to garbage collect them. I haven't looked into
it much though, it seems to work.  Sorry I cannot answer your
question better.
--
  Henry Miller
  hank@millerfarm.com



On Mon, Sep 11, 2017, at 11:37 AM, Eric Chamberland wrote:
> Ok I have set ICECC_DEBUG=debug.


> Looking at the logs, I can see it is working right now:


> ... [18022] 12:11:04: handle_transfer_env [18022] 12:11:04: got pid
> 18343 [18022] 12:11:05: handle_transfer_env_done [18022] 12:11:05:
> installed_size: 87289375 [18022] 12:11:05: installed
> x86_64/a56529631d0633503a3a54be0af992f7 size: 87289375 all: 173484305
> [18022] 12:11:05: reannounce_environments
>> But I am asking myseld: since the very same "environment" is
> transfered, when is it deleted or retransfered?  Can it be deleted by
> a cleanup process (garbage collection) just while I am sending a new
> request?> Thanks,
>  Eric


> — You are receiving this because you were mentioned. Reply to this
> email directly, view it on GitHub[1], or mute the thread[2].> 

Links:

  1. https://github.com/icecc/icecream/issues/184#issuecomment-328586044

**Comment by ericch1** [comment]
I have added ICECC_DEBUG=debug, but on the faulty workstation, I still have no relevant new error written.

Here is what I have today:
<pre>
[29246] 11:45:32: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668752
[29247] 11:45:32: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668754
[29248] 11:45:32: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668755
[29249] 11:45:32: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668765
[29250] 11:45:32: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668766
[29251] 11:45:33: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668768
[29252] 11:45:33: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668769
[29253] 11:45:33: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668770
[29254] 11:45:33: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668771
[29255] 11:45:33: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668772
[29256] 11:45:33: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668773
[29257] 11:45:33: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668774
[29258] 11:45:35: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668782
[29259] 11:45:35: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668786
[29260] 11:45:35: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668788
[29261] 11:45:35: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668796
[29262] 11:45:35: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668798
[29263] 11:45:35: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668800
[29264] 11:45:35: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668799
[29265] 11:45:35: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668797
[29266] 11:45:35: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668809
[29267] 11:45:36: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668812
[29268] 11:45:36: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668814
[29269] 11:45:36: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668813
[29270] 11:45:36: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668815
[29271] 11:45:36: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668817
[29272] 11:45:36: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668816
[29273] 11:45:36: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668818
[29274] 11:45:36: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1668819
[29275] 11:45:40: I don't have environment a56529631d0633503a3a54be0af992f7(x86_64) 1669365
[5320] 11:45:41: handle_transfer_env
[5320] 11:45:41: got pid 29276
[5320] 11:45:44: handle_transfer_env_done
[5320] 11:45:44: installed_size: 77549582
[5320] 11:45:44: installed x86_64/ICECC_VERSION_clang3.8 size: 77549582 all: 103361547
[5320] 11:45:44: reannounce_environments 
[5320] 11:51:21: handle_transfer_env
[5320] 11:51:21: got pid 30251
[5320] 11:51:23: handle_transfer_env_done
[5320] 11:51:23: installed_size: 87289375
[5320] 11:51:23: installed x86_64/a56529631d0633503a3a54be0af992f7 size: 87289375 all: 190650922
[5320] 11:51:23: reannounce_environments 
</pre>

Maybe it is on the "client" side I should add  ICECC_DEBUG=debug?

Thanks for your help!

Eric

**Comment by ericch1** [comment]
Doh!
I just found that all workstations and scheduler where using the latest git version, *but*, I was launching the "old" /usr/bin/icecc  wrapper... :/
/usr/bin/icecc --version                                                                                                                                                      
ICECC 1.0.1

can it explain something?

Should it be forbidden?

Thanks,
Eric

**Comment by ericch1** [comment]
New Informations!!!!

I still got the issue, but now I have more logs... :)

Here is the log output on the client workstation (with iceccd -vvv):
<pre>
...
[166923] 17:27:29: timeout <= 0
[166923] 17:27:29: !wait_for_msg()
...
[166257] 13:26:19: scheduler->send_msg( JobLocalDoneMsg( 13) );
[166257] 13:26:40: accepted 11 local unix domain socket
[166257] 13:26:40: send JobLocalBeginMsg to client
[166257] 13:26:40: pushed local job 14
[166257] 13:26:40: saw eof without complete msg! 1
</pre>

and the almost usual error from the wrapper:
<pre>
ICECC[154472] 13:09:19: flush_writebuf() failed Broken pipe
ICECC[154472] 13:09:19: Remote status (compiled on 132.203.7.21): /var/cache/icecream/target=x86_64/a56529631d0633503a3a54be0af992f7/usr/bin/as is not executable
ICECC[154472] 13:09:19: got exception Error 23 - Remote status (compiled on 132.203.7.21)
/var/cache/icecream/target=x86_64/a56529631d0633503a3a54be0af992f7/usr/bin/as is not executable (132.203.7.21) 
</pre>

and almost nothing on remote side:
<pre>
[7430] 13:29:50: requests--98430
[7430] 13:29:50: handle connection returned 9989
</pre>


Doh, look at this odd thing:
<pre>
[166257] 13:28:55: send JobLocalBeginMsg to client
[166257] 13:28:55: pushed local job 37
[166257] 13:29:00: saw eof without complete msg! 1
[166257] 13:29:00: scheduler->send_msg( JobLocalDoneMsg( 37) );
[166257] 13:29:01: accepted 11 132.203.7.54
[166257] 13:29:01: requests--97621
[166257] 13:29:01: handle connection returned 166991
[166257] 13:29:01: accepted 13 132.203.7.54
[166257] 13:29:01: accepted 14 132.203.7.54
[166257] 13:29:01: requests--97623
[166257] 13:29:01: handle connection returned 166994
[166257] 13:29:01: requests--97624
[166257] 13:29:01: handle connection returned 166996
[166257] 13:29:01: accepted 17 132.203.7.202
[166257] 13:29:01: handle_transfer_env
[166257] 13:29:01: pid 167000
[166257] 13:29:01: got pid 167000
[166994] 17:29:01: <parent, waiting>
[166994] 17:29:01: timeout <= 0
[166994] 17:29:01: !wait_for_msg()
[166994] 17:29:02: timeout <= 0
[166994] 17:29:02: !wait_for_msg()
[166994] 17:29:02: timeout <= 0
</pre>

some of the wallclock times are in the futur!!!???

Strange???

Hope this help!

Eric

**Comment by estebanpapp** [comment]
I am having this issue frequently in a small network with mostly MacOS nodes

The problem in my case is that some machines removed /tmp/icecc-envs. MacOS removes files/folders from /tmp that have not been accessed for three days, some more info: https://superuser.com/questions/187071/in-macos-how-often-is-tmp-deleted

Looks like the daemon doesnt realize that the environment was removed. 
I can reproduce the issue by just removing the /tmp/icecc-envs in one of the machines while the daemon is running.

In my case, I am going to configure the nodes to pass the parameter: "env-basedir" pointing to a place that doesnt get periodically removed. 

Some better, long-term, potential solutions:
- re-announce environments when this error happens in a node
- (preferred) put a file watcher on /tmp/icecc-envs and re-announce environments when the folder is updated
- "touch" the environment files frequently

**Comment by llunak** [comment]
Installed environments are icecream's files and nobody from the outside is supposed to just randomly remove them (and I don't even think that's a clean way to detect this or recover). So I've changed the default location to use /var/tmp instead and the error message now should hint at the problem, but if you have this problem, just use a safe location.

**Comment by dholbert** [comment]
I hit this error fairly frequently, for what it's worth -- often after I've interrupted a build job by doing Ctrl+C (which I do fairly often, if I think of some change I need to make etc.)

My suspicion is that things go wrong when my "Ctrl+C" action lands in the middle of icecc's transfer-toolchain-to-remote-host process, for some remote host.  I think this results in the remote host being left with an incomplete tarball -- and when I start building again, any jobs distributed to that remote host will trigger this error output, because the host thinks it has the toolchain, but then it fails to extract it.

**Comment by dholbert** [comment]
@llunak as noted in my comment above, it seems like we can easily get into this situation *without* other processes messing with icecc's files, if a toolchain-transfer is simply interrupted by a Ctrl+C or by a network disconnection. Any thoughts on how to gracefully recover from this scenario?

(Let me know if I should file a new issue; I'm not sure whether [your comment above](https://github.com/icecc/icecream/issues/184#issuecomment-403204436) was meant as a "wontfix" [in which case I'm hoping my new scenario might make you reconsider], or as a "landed some code to fix the part that we could identify" [in which case maybe a new issue is appropriate])

**Comment by llunak** [comment]
I have improved error handling related to environment transfer, so this should be fixed now. If somebody still has this problem with the upcoming 1.3 version, please create a new issue.
